From: Jason Rhinelander <jason@imaginary.ca>
Date: Sat, 14 Dec 2019 11:36:58 -0400
Subject: net/if.h hack for xenial

---
 include/tuntap.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/tuntap.h b/include/tuntap.h
index 5dff088..6b37e7b 100644
--- a/include/tuntap.h
+++ b/include/tuntap.h
@@ -36,7 +36,12 @@
 #include <netinet/in.h>
 #endif
 
+#if defined(Linux)
+// Once we drop xenial support we can just include net/if.h on linux
+#include <linux/if.h>
+#else
 #include <net/if.h>
+#endif
 
 #if defined Linux
 #include <netinet/in.h>
